### PR TITLE
Renaming `WeaveTensorGraph` to `WeaveTensorModel`

### PR DESCRIPTION
**Pull request by mlgill**
This branch will be used for renaming `WeaveTensorGraph` to `WeaveTensorModel`, per the issue [here](https://github.com/deepchem/deepchem/issues/1138).

### CODE REVIEW

**Comment by rbharath** [comment]
@mlgill This looks great! We should be good to merge in once you resolve the merge conflicts. You can try the GitHub web editor, or can alternatively `git pull origin master` on your local repo, manually edit the conflict files, commit and push the updated version to the PR.

**Comment by mlgill** [comment]
@rbharath Done -- the CI build is running now.

File: deepchem/models/__init__.py [hunk 1]
```diff
@@ -16,8 +16,8 @@
 from deepchem.models.tensorgraph.IRV import TensorflowMultiTaskIRVClassifier
 from deepchem.models.tensorgraph.robust_multitask import RobustMultitaskClassifier
 from deepchem.models.tensorgraph.robust_multitask import RobustMultitaskRegressor
-from deepchem.models.tensorgraph.progressive_multitask import ProgressiveMultitaskRegressor, ProgressiveMultitaskClassifier
-from deepchem.models.tensorgraph.models.graph_models import WeaveTensorGraph, DTNNTensorGraph, DAGTensorGraph, GraphConvModel, MPNNTensorGraph
+from deepchem.models.tensorgraph.progressive_multitask import ProgressiveMultitaskRegressor
```
Comment (rbharath) [inline deepchem/models/__init__.py]: Sorry, missed this on first pass. Could you add the `ProgressiveMultitaskClassifier` import back in?

File: deepchem/models/tensorgraph/models/graph_models.py [hunk 2]
```diff
@@ -1149,13 +1149,24 @@ def predict_on_generator(self, generator, transformers=[]):
 
 import warnings
 
+TENSORGRAPH_DEPRECATION = "{} is deprecated and has been renamed to GraphConvModel and will be removed in DeepChem 3.0."
```
Comment (rbharath) [inline deepchem/models/tensorgraph/models/graph_models.py]: This is really nice! Good design choice :)

File: deepchem/models/__init__.py (hunk 1 above)
Comment (mlgill) [reply deepchem/models/__init__.py]: Done and pushed -- sorry about that. I suspect it was a stray keystroke as I was editing on a plane.

File: deepchem/models/tensorgraph/models/graph_models.py (hunk 2 above)
Comment (mlgill) [reply deepchem/models/tensorgraph/models/graph_models.py]: That has a bug with the second title though. Fixing now.

Comment (mlgill) [reply deepchem/models/tensorgraph/models/graph_models.py]: Done.

**Comment by mlgill** [comment]
@rbharath  Looks like the 3.5 tests passed but the 2.7 did not. I've dug through the logs and am not entirely sure where the failure is (also, I can't think of any of my changes that would have been 2.7 incompatible). I'll hunt more later, but any guidance would be appreciated.

**Comment by peastman** [comment]
I think it was just a rare stochastic failure.  I've restarted the 2.7 tests.

**Comment by rbharath** [comment]
@mlgill I think this looks good to merge in! If there are no further changes you'd like to make, could you remove the [WIP] tag?

**Comment by mlgill** [comment]
@rbharath I'm all done --removed the [WIP] tag.

**Comment by rbharath** [comment]
@mlgill Thanks!

LGTM

**Pull request merged**